ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -1319,6 +1326,52 @@ void persistIndex() throws StoreException {
   }
 
   /**
+   * Re-validates the {@code token} to ensure that it is consistent with the current set of index segments. If it is
+   * not, a {@link StoreFindToken.Type#Uninitialized} token is returned.
+   * @param token the {@link StoreFindToken} to revalidate.
+   * @return {@code token} if is consistent with the current set of index segments, a new
+   * {@link StoreFindToken.Type#Uninitialized} token otherwise.
+   */
+  FindToken revalidateFindToken(FindToken token) {
```

Comment:
do we need to check for revalidation for getBlobReadInfo(). I mean, once we get the indexValue, is there a chance that the log entry was compacted?